tests/conftest.py

```diff
@@ -15,10 +19,25 @@ def __init__(self, app: Typer, test_snowcli_config: str):
     @functools.wraps(CliRunner.invoke)
     def invoke(self, *a, **kw):
         kw.update(catch_exceptions=False)
-        return super().invoke(self.app, *a, **kw)
-
-    def invoke_with_config(self, *args, **kwargs):
-        return self.invoke(
-            ["--config-file", self.test_snowcli_config, *args[0]],
-            **kwargs,
+        return super().invoke(
+            self.app, ["--config-file", self.test_snowcli_config, *a[0]], **kw
         )
+
+
+@pytest.fixture(autouse=True)
+def reset_global_context_after_each_test(request):
+    snow_cli_global_context.reset_global_context()
+    yield
+
+
+# This cleanup function is required to avoid random breaking of logging
+# in one test caused by presence of capsys in other test.
+# See similar issues: https://github.com/pytest-dev/pytest/issues/5502
+@pytest.fixture(autouse=True)
+def clean_logging_handlers(request):
```

Comment:
Is has to be added to fix logging issues started to being visible after introducing random test order.